src/promql/src/functions/aggr_over_time.rs

```diff
@@ -117,7 +117,31 @@ pub fn present_over_time(_: &TimestampMillisecondArray, values: &Float64Array) -
     }
 }
 
-// TODO(ruihang): support quantile_over_time, stddev_over_time, and stdvar_over_time
+// TODO(ruihang): support quantile_over_time, and stdvar_over_time
+
+/// the population standard deviation of the values in the specified interval.
+#[range_fn(
+    name = "StddevOverTime",
+    ret = "Float64Array",
+    display_name = "prom_stddev_over_time"
+)]
+pub fn stddev_over_time(_: &TimestampMillisecondArray, values: &Float64Array) -> Option<f64> {
+    if values.is_empty() {
+        None
+    } else {
+        let count = values.len();
+        let sum = compute::sum(values);
+        let sum_sq = compute::sum(
+            &values
+                .into_iter()
+                .map(|val| val.map(|v| v * v))
+                .collect::<Float64Array>(),
+        );
```

Comment:
I thought of using `for` loop and mutable bindings to create `sum` and `sum_sq`. wdyt?

Reply:
Both look good to me!  The only thing I'd like to argue is the collected type. `Float64Array` is a bit more expensive than plan `Vec<f64>`. You can try `values.values().iter().sum()` and `values.values().iter().map(|v| v*v).sum()` to avoid `_Array`